src/SingleSignatureDirtyHack.sol

```diff
@@ -187,6 +195,63 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
         }
     }
 
+    function _encodePermitHash(IERC20 token, address from, uint256 nonce, uint256 sellAmount, bytes32 signingHash)
+        private
+        view
+        returns (bytes32 result)
+    {
+        // keccak256(abi.encodePacked(
+        //    hex"1901",
+        //    token.DOMAIN_SEPARATOR(),
+        //    keccak256(abi.encode(
+        //             keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"),
+        //             owner,
+        //             spender,
+        //             value,
+        //             nonce,
+        //             deadline))
+        // ))
+        bytes32 permitTypeHash = PERMIT_TYPEHASH;
+        assembly ("memory-safe") {
+            let ptr1 := mload(0x40)
+            let ptr2 := mload(0x80)
+            let ptr3 := mload(0xa0)
+
+            mstore(0x00, permitTypeHash)
+            mstore(0x20, from)
+            mstore(0x40, address())
+            mstore(0x60, sellAmount)
+            mstore(0x80, nonce)
+            mstore(0xa0, signingHash)
```

Comment:
you must never clobber the memory in the range `0x80` to `mload(0x40)`. that is memory reserved by the compiler for spilling variables from the stack